Roll to Playwright v1.37.1

canstand opened the pull request:
Rolling to playwright v1.37.1 and several bug fixes while improving test coverage.

mxschmitt commented on .github/workflows/build.yml:
```diff
@@ -36,6 +36,21 @@ jobs:
+          ~/Library/Caches/ms-playwright
+          ~\AppData\Local\ms-playwright-go
+          ~\AppData\Local\ms-playwright
+        key: ${{ runner.os }}-go-${{ matrix.browser }}-${{ hashFiles('**/package.json') }}
```
Hashing the package.json feels wrong here, I guess it should be run.go instead?

canstand replied:
actions/setup-go@v4 already caches go modules.  actions/cache is only used to cache the playwright driver and browsers, so only the playwright version in `package.json` is required.

mxschmitt replied:
but which package.json file do you target? Since this project only has a package.json inside the driver, aka. the ms-playwright-go directory if I remember correctly.

canstand replied:
Oh, I misunderstood, thought `cache/msplaywrigh-go/package.json` was referenced here. change it now.

mxschmitt commented on artifact.go:
```diff
@@ -39,24 +39,36 @@ func (a *artifactImpl) SaveAs(path string) error {
 	return err
 }
 
+func (a *artifactImpl) ReadInfoBuffer() ([]byte, error) {
```
```suggestion
func (a *artifactImpl) ReadIntoBuffer() ([]byte, error) {
```

canstand commented:
Hi @mxschmitt, could you please tag a new version?

mxschmitt commented:
@canstand do you mean permission wise it won't work or do you prefer if I create releases? This release has breaking changes, so it might be worth if you prepare the release notes, would appreciate it a lot!

Version wise it should be v0.3700.0 I believe. The extra .0 at the end is important so the Go version system picks it up.

canstand commented:
Currently I do not have permission to create tags. I can do it if you agree. 
Agree that the tag remains `v0.xxyy.z`, `xx` corresponds to the playwright version, `yy.z` for bug fixes.

Release notes can be based on the previous pr message, for reference:
```md
This version is on pair driver-wise with the Playwright 1.37 version.

BREAKING CHANGE: 
Due to the rewrite of the go code generation scripts, it brings a lot of changes. 
Note the following 4 types:

1. Optimized generated comments. Links and **Deprecated** tags in Go doc comments now work.
  - May cause many **Deprecated** lint errors, please update the call or use `//nolint:staticcheck` to ignore.
1. Added event interface methods. For example `WebSocket.OnFrameReceived()` etc. 
1. The signatures of some interface methods have changed, for example:
  - Unified optional parameter naming. For example, previously all `Click` used `PageClickOptions`, now
    ```go
    Locator.Click(options ...LocatorClickOptions)
    Page.Click(selector string, options ...PageClickOptions)
    Frame.Click(selector string, options ...FrameClickOptions)
    ```
  - Some parameters are generated as new types or names, e.g. `Page.AddInitScript(script Script)`
1. Removed several methods that were not documented upstream. These methods can be safely converted to other existing methods.  For example
  - `Page.ExpectedDialog` can use `Page.OnDialog` instead.
  - All `xxxAssertions.NotTo...`, use `.Not().xxx` pls.
  - `WebSocket.On("framesent", ...), use `WebSocket.OnFrameSent(...).
```

mxschmitt commented:
@canstand should be possible now, I changed the role from Write to Maintain, thanks!
